src/api/wayland/keyboard.rs

```diff
@@ -37,8 +38,19 @@ impl wayland_kbd::Handler for KbdHandler {
             };
             let vkcode = key_to_vkey(rawkey, keysym);
             let mut guard = eviter.lock().unwrap();
-	    // TODO implement ModifiersState
-            guard.push_back(Event::KeyboardInput(state, rawkey as u8, vkcode, ModifiersState::default()));
+            guard.push_back(
+                Event::KeyboardInput(
+                    state,
+                    rawkey as u8,
+                    vkcode,
+                    ModifiersState {
+                        shift: mods.shift ^ mods.caps_lock,
```

Comment:
Actually, the X11 backend does not merge shift and caps lock modifiers. This is IMO sensible, as most keymaps on linux react differently to shift and caps lock.

However, the X11 backend here simply discards the caps lock state... @rigtorp did you chose this same behaviour for all platforms?